src/epd2in9b/mod.rs

```diff
@@ -0,0 +1,322 @@
+        self.send_resolution(spi)?;
+
+        let color = DEFAULT_BACKGROUND_COLOR.get_byte_value();
+
+        // Clear the black
+        self.interface
+            .cmd(spi, Command::DATA_START_TRANSMISSION_1)?;
+
+        self.interface.data_x_times(spi, color, NUM_DISPLAY_BITS)?;
+
+        // Clear the red
+        self.interface
+            .cmd(spi, Command::DATA_START_TRANSMISSION_2)?;
+        self.interface.data_x_times(spi, color, NUM_DISPLAY_BITS)?;
+
+        self.wait_until_idle();
+        Ok(())
+    }
+
+    fn set_lut(
+        &mut self,
+        _spi: &mut SPI,
+        _refresh_rate: Option<RefreshLUT>,
+    ) -> Result<(), SPI::Error> {
+        Ok(())
+    }
+
+    fn is_busy(&self) -> bool {
+        self.interface.is_busy(IS_BUSY_LOW)
+    }
+}
+
+impl<SPI, CS, BUSY, DC, RST> EPD2in9b<SPI, CS, BUSY, DC, RST>
+where
+    SPI: Write<u8>,
+    CS: OutputPin,
+    BUSY: InputPin,
+    DC: OutputPin,
+    RST: OutputPin,
+{
+    fn command(&mut self, spi: &mut SPI, command: Command) -> Result<(), SPI::Error> {
+        self.interface.cmd(spi, command)
+    }
+
+    fn send_data(&mut self, spi: &mut SPI, data: &[u8]) -> Result<(), SPI::Error> {
+        self.interface.data(spi, data)
+    }
+
+    fn cmd_with_data(
+        &mut self,
+        spi: &mut SPI,
+        command: Command,
+        data: &[u8],
+    ) -> Result<(), SPI::Error> {
+        self.interface.cmd_with_data(spi, command, data)
+    }
+
+    fn wait_until_idle(&mut self) {
+        self.interface.wait_until_idle(IS_BUSY_LOW)
+    }
+
+    fn send_resolution(&mut self, spi: &mut SPI) -> Result<(), SPI::Error> {
+        let w = self.width();
+        let h = self.height();
+
+        self.command(spi, Command::RESOLUTION_SETTING)?;
+
+        self.send_data(spi, &[w as u8])?;
+        self.send_data(spi, &[(h >> 8) as u8])?;
+        self.send_data(spi, &[h as u8])
+    }
+
+    /// Set the outer border of the display to the chosen color.
+    pub fn set_border_color(&mut self, spi: &mut SPI, color: TriColor) -> Result<(), SPI::Error> {
+        let border = match color {
+            TriColor::Black => BLACK_BORDER,
+            TriColor::White => WHITE_BORDER,
+            TriColor::Red => RED_BORDER,
+        };
+        self.cmd_with_data(
+            spi,
+            Command::VCOM_AND_DATA_INTERVAL_SETTING,
+            &[border | VCOM_DATA_INTERVAL],
+        )
+    }
+
+    /// Update only the black/white data of the display.
+    ///
+    /// Finish by calling `update_red_frame`.
+    pub fn update_mono_frame(&mut self, spi: &mut SPI, black: &[u8]) -> Result<(), SPI::Error> {
+        self.interface
+            .cmd(spi, Command::DATA_START_TRANSMISSION_1)?;
+        self.interface.data(spi, black)?;
+        Ok(())
+    }
+
+    /// Update only red data of the display.
+    ///
+    /// This data takes precedence over the black/white data.
+    pub fn update_red_frame(&mut self, spi: &mut SPI, red: &[u8]) -> Result<(), SPI::Error> {
```

Comment:
rename it to update_color_frame or similar please and try adding it to the threecolordisplay

Reply:
`update_color_frame` exists already for `ThreeColorDisplay` and updates both at the same time. 
I also find it a better name though for this case than for mono + color, could the existing one be renamed? Otherwise `update_only_color_frame`... naming things is hard

Reply:
Yes this makes it more concise :+1: